packages/fiori/src/Timeline.ts

```diff
@@ -111,6 +190,112 @@ class Timeline extends UI5Element {
 			: Timeline.i18nBundle.getText(TIMELINE_ARIA_LABEL);
 	}
 
+	get showBusyIndicatorOverlay() {
+		return !this.growsWithButton && this.loading;
+	}
+
+	get growsOnScroll(): boolean {
+		return this.growing === TimeLineGrowingMode.Scroll;
+	}
+
+	get growingButtonIcon() {
+		return this.layout === TimelineLayout.Horizontal ? process : drillDown;
+	}
+
+	get growsWithButton(): boolean {
+		return this.growing === TimeLineGrowingMode.Button;
+	}
+
+	onAfterRendering() {
+		if (this.growsOnScroll) {
+			this.observeTimeLineEnd();
+		} else if (this.timeLineEndObserved){
+			this.unobserveTimelineEnd();
+		}
+
+		this.growingIntersectionObserver = this.getIntersectionObserver();
+	}
+
+	onExitDOM() {
+		this.unobserveTimelineEnd();
+	}
+
+	observeTimeLineEnd() {
+		if (!this.timeLineEndObserved) {
+			this.getIntersectionObserver().observe(this.timelineEndMarker);
+			this.timeLineEndObserved = true;
+		}
+	}
+
+	async observeTimelineEnd() {
+		if (!this.timeLineEndObserved) {
+			//await renderFinished();
+			this.getIntersectionObserver().observe(this.timelineEndDOM!);
+			this.timeLineEndObserved = true;
+		}
+	}
+
+	unobserveTimelineEnd() {
+		if (this.growingIntersectionObserver) {
+			this.growingIntersectionObserver.disconnect();
+			this.growingIntersectionObserver = null;
+			this.timeLineEndObserved = false;
+		}
+	}
+
+	get timelineEndDOM() {
+		return this.shadowRoot!.querySelector(".ui5-timeline-end-marker");
```

Comment:
the best practice is to use `@query` for this, but it's ok to leave it, just FYI

Reply:
Done